File: integration-tests/grpc-domain-socket/src/main/java/io/quarkus/it/grpc/uds/HelloWorldService.java
package io.quarkus.it.grpc.uds;

import io.quarkus.grpc.GrpcService;
import io.smallrye.mutiny.Uni;

@GrpcService
public class HelloWorldService extends MutinyGreeterGrpc.GreeterImplBase {

    @Override
    public Uni<HelloReply> sayHello(HelloRequest request) {
        String name = request.getName();
        return Uni.createFrom().item(
                HelloReply.newBuilder().setMessage("Hello " + name).build());
    }
}

File: integration-tests/grpc-kotlin-coroutine/src/main/kotlin/io/quarkus/grpc/examples/hello/HelloWorldService.kt
package io.quarkus.grpc.examples.hello

import examples.HelloReply
import examples.HelloRequest
import io.quarkus.grpc.GrpcService
import io.smallrye.common.vertx.VertxContext
import io.vertx.core.Vertx

@GrpcService
class HelloWorldService : GreeterCoroutineImplBase() {

    override suspend fun sayHello(request: HelloRequest): HelloReply {
        val ctx = Vertx.currentContext()
        val isDuplicated = ctx != null && VertxContext.isDuplicatedContext(ctx)

        return HelloReply.newBuilder()
            .setMessage("Hello ${request.name}")
            .setHasVertxContext(ctx != null)
            .setIsDuplicatedContext(isDuplicated)
            .build()
    }
}
